src/lib.rs

```diff
@@ -57,11 +58,37 @@ pub trait Number: Sized {
     /// Maximum value that can be represented by this type
     const MAX: Self;
 
+    /// Creates a number from the given value, throwing an error if the value is too large
```

Comment:
do you want to mention that the argument has to be the underlying type, i.e. the next power of two number of bits, which is fine in case your data is already exactly this, otherwise you should use `new_u32()` etc.?